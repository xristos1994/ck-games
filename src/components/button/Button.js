import React from "react";
import { classnames } from "@utils/component-utils";
import styles from "./styles.module.css";

const Button = ({ children, onClick, className = "", ...other }) => {
  return (
    <button
      className={classnames(styles.button, className)}
      {...other}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

export { Button };
