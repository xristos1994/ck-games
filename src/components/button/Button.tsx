import React, { FC, MouseEventHandler, ReactElement, ReactChild } from "react";
import { classnames } from "@utils/component-utils";
import "./styles.module.css";

interface IProps {
  children: ReactChild | ReactChild[];
  onClick?: MouseEventHandler<HTMLButtonElement>;
  className: string;
  other?: Record<string, unknown>;
}

const Button: FC<IProps> = ({
  children,
  onClick,
  className = "",
  other,
}): ReactElement => {
  return (
    <button
      className={classnames("main-button-hover-effect", {
        [className]: !!className,
      })}
      {...other}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

export { Button };
