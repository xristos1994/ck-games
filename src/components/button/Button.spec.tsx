import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from './Button';

describe('components/Button', () => {
  it('renders correctly', () => {
    let tree = renderer.create(
      <Button>
        Test Label
      </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <Button className="Test-class-name">
        Test Label
      </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <Button className="Test-class-name" other={{ disabled: true }}>
        Test Label
      </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <Button className="Test-class-name" other={{ disabled: true }} onClick={() => console.log('Button Clicked')}>
        Test Label
      </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
