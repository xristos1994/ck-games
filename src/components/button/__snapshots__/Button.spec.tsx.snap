// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button renders correctly 1`] = `
<button
  className="main-button-hover-effect"
>
  Test Label
</button>
`;

exports[`components/Button renders correctly 2`] = `
<button
  className="main-button-hover-effect Test-class-name"
>
  Test Label
</button>
`;

exports[`components/Button renders correctly 3`] = `
<button
  className="main-button-hover-effect Test-class-name"
  disabled={true}
>
  Test Label
</button>
`;

exports[`components/Button renders correctly 4`] = `
<button
  className="main-button-hover-effect Test-class-name"
  disabled={true}
  onClick={[Function]}
>
  Test Label
</button>
`;
